### ISSUE TITLE
MLP requests failing after merging PR #67

### ISSUE THREAD

**Issue by FerUy** [RestComm/gmlc#69]
GET requests works fine (e.g. http://127.0.0.1:8080/restcomm/gmlc/rest?msisdn=577438634) but when trying MLP request intended for success (not dummy number), we are getting this error:

$ curl -X POST -d @mlpreq.txt http://192.168.26.128:8380/restcomm/gmlc/mlp
`<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE svc_result SYSTEM "MLP_SVC_RESULT_310.DTD">
<svc_result xmlns="MLP_SVC_RESULT_310.dtd" ver="3.1.0">
    <slia ver="3.1.0">
        <result resid="1">SYSTEM FAILURE</result>
        <add_info>IO failure generating XML</add_info>
    </slia>
</svc_result>`

Syncing with master is needed before fixing this, as PR #67 merging was done previously to fixing issue #8 (PR #68), hence it overlaps with some of the classes in which this work was done (especially MobileCoreNetworkInterfaceSbb.java)

**Comment by FerUy** [comment]
@Monix, @angrygreenfrogs, just created this issue to solve what I just mentioned happened after merging PR #67 . Again, please sync with master before investigating, as some classes have been updated in today's merge of PR #68, like MobileCoreNetworkInterfaceSbb.java (where I believe the problem is generated).

**Comment by Vanit** [comment]
@FerUy Will look into this today!

**Comment by FerUy** [comment]
Great, thanks @Monix !

**Comment by Vanit** [comment]
@FerUy I've merged the upstream changes and I've been unable to reproduce the issue in my existing build and a new build.

It could be that I'm being blocked by a separate issue - whenever I've made non-fake requests I get a 501 unsupported error from somewhere in the SCCP stack, and have to read the DialogTimeout failure from my jboss output. I assumed this behaviour was normal since I don't have anything configured.

What setup are you using for testing this? Simulator?

**Comment by FerUy** [comment]
Correct @Monix, in **simulator** mode.

**Comment by Vanit** [comment]
@FerUy Fixed the issue, the CGIResponse defaults weren't XML friendly. Now MLP OK will return -1 for the location fields instead of failing.

**Comment by FerUy** [comment]
Confirmed @Monix, well done, all tests are now working properly. Sorry for my absence since opening this issue, but been overwhelmed with other support/training content activities. Anyway, I was sure you'd solve it easily as you did ;)
Thanks!

**Comment by Vanit** [comment]
@FerUy No problem! Was my bad for not testing with the simulator :)
